src/gyroaverage/gyroaverage_operator.hpp

```diff
@@ -0,0 +1,213 @@
+    /**
+     * @brief Constructor.
+     * @param[in] rho_L Field of Larmor radii on the (r, theta) grid.
+     */
+    explicit GyroAverageOperator(DFieldRTheta const& rho_L) : m_rho_L(rho_L) {}
+
+    /**
+     * @brief Applies the gyroaverage operator to a batched field.
+     *
+     * For each batch, and for each (r, theta) grid point, computes the gyroaverage by
+     * integrating the field along a circle of radius rho_L centred at (r, theta).
+     * The field is interpolated at off-grid points using 2D B-splines.
+     *
+     * @tparam CoordinateTransformFunction
+     *         Callable type that transforms (R, Z) coordinates to (r, theta) coordinates.
+     * @param[in] A Input field to be gyroaveraged (batched).
+     * @param[out] A_bar Output field to store the gyroaveraged result (batched).
+     * @param[in] coordinate_transform Function to convert (R, Z) to (r, theta).
+     * @param[in] nb_gyro_points Number of points to use in the gyroaverage integration (default: 8).
+     */
+    template <class CoordinateTransformFunction>
+    void operator()(
+            DFieldRThetaBatch const& A,
+            DFieldRThetaBatch const& A_bar,
+            CoordinateTransformFunction coordinate_transform,
+            std::size_t nb_gyro_points = 8) const
+    {
+        static_assert(
+                std::is_invocable_v<CoordinateTransformFunction, double, double>,
+                "CoordinateTransformFunction must be a functor on (double, double)");
+        IdxRangeR const r_domain = get_idx_range<GridR>(A);
+        IdxRangeTheta const theta_domain = get_idx_range<GridTheta>(A);
+        IdxRangeBatch const batch_domain = get_idx_range<GridBatch>(A);
+        IdxRangeRTheta const rtheta_mesh = get_idx_range<GridR, GridTheta>(A);
+
+        ddc::NullExtrapolationRule r_extrapolation_rule;
+        ddc::PeriodicExtrapolationRule<DimensionThetaType> theta_extrapolation_rule;
+
+        SplineRThetaBuilder const spline_builder(rtheta_mesh);
+        SplineRThetaEvaluatorNullBound const spline_evaluator(
+                r_extrapolation_rule,
+                r_extrapolation_rule,
+                theta_extrapolation_rule,
+                theta_extrapolation_rule);
+
+        // Instantiate chunk of spline coefs to receive output of spline_builder (r, theta)
+        DFieldMemBSRTheta coef_alloc(
+                spline_builder.batched_spline_domain(rtheta_mesh),
+                ddc::DeviceAllocator<double>());
+        DFieldBSRTheta const coef = get_field(coef_alloc);
+        DConstFieldRTheta const rho_L = get_const_field(m_rho_L);
+
+        using SubDFieldRTheta = DField<
+                IdxRangeRTheta,
+                typename ExecutionSpace::memory_space,
+                Kokkos::layout_stride>;
+
+        ddc::for_each(batch_domain, [&](IdxBatch const ib) {
+            // FIXME
+            // The input of the spline builder must be LayoutRight
+            // We allocate a buffer in LayoutRight whereto the slice is copied
```

Comment:
What needs to be fixed ? Avoid an allocation for each batch iteration ?

Does Gyselalibxx work with `FIXME` in the sources ?

Reply:
Ok. S then what needs to be fixed?

Reply:
I don't consider something that needs looking at for optimisation to be broken (as fixme implies). I would rather a dedicated issue instead of this comment